src/App.tsx

```diff
@@ -1,26 +1,140 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import React, { useEffect, useRef, useState } from 'react';
+import { Todo } from './types/Todo';
+import * as apiService from './api/todos';
+import { TodoHeader } from './components/TodoHeader';
+import { TodoList } from './components/TodoList';
+import { TodoFooter } from './components/TodoFooter';
+import { ErrorNotifications } from './components/ErrorNotifications';
+import { TypeFilter } from './types/TypeFilter';
+import { getFilteredTodos } from './utils/filterTodos';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
+  const inputRef = useRef<HTMLInputElement | null>(null);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+
+  const [newTodoInput, setNewTodoInput] = useState('');
+
+  const [errorMessage, setErrorMessage] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+  const [loadingIds, setLoadingIds] = useState<number[]>([]);
+  const [filterBy, setFilterBy] = useState(TypeFilter.All);
+
+  const notCompletedTasksCounter = todos.filter(todo => !todo.completed).length;
+  const completedTasks = todos.filter(todo => todo.completed);
+  const filteredTodos = getFilteredTodos(todos, filterBy);
+
+  const loadTodos = () => {
+    setErrorMessage('');
+    apiService
+      .getTodos()
+      .then(setTodos)
+      .catch(() => setErrorMessage('Unable to load todos'))
+      .then(() => {
+        setTimeout(() => {
+          setErrorMessage('');
+        }, 3000);
+      });
+  };
+
+  useEffect(() => {
+    loadTodos();
+  }, []);
+
+  function addTodo(event: React.FormEvent<HTMLFormElement>) {
+    event.preventDefault();
+    const title = newTodoInput.trim();
+
+    if (title) {
+      setIsLoading(true);
+
+      const userId = apiService.USER_ID;
+      const completed = false;
+
+      setTempTodo({ id: 0, title, userId, completed });
+
+      apiService
+        .createTodo({ userId, title, completed })
+        .then(newTodo => {
+          setTodos(currentTodos => [...currentTodos, newTodo]);
+          setNewTodoInput('');
+        })
+        .catch(() => setErrorMessage('Unable to add a todo'))
+        .finally(() => {
+          setTempTodo(null);
+          setIsLoading(false);
+        });
+    } else {
+      setErrorMessage('Title should not be empty');
+    }
+  }
+
+  function handleDeleteTodo(todoIds: number[]) {
```

Comment:
You need to stick to one way of creating functions, fix it everywhere
```suggestion
  const handleDeleteTodo = (todoIds: number[]) => {
```